### PR TITLE
Update outdated objects

### LINKED ISSUE #34: update outdated objects
update outdated objects

### PR DESCRIPTION
**Pull request by outmaneuver**
Fixes #34

Add methods to update outdated objects in various classes.

* **`discmoji/_http.py`**: Add `update_object` method in `EndpointManager` class to update outdated objects.
* **`discmoji/bot.py`**: Add `update_all_objects` method in `Bot` class to update all outdated objects.
* **`discmoji/channel.py`**: Add `update` method in `GuildTextChannel` class to update outdated objects.
* **`discmoji/command.py`**: Add `update` method in `Command` class to update outdated objects.
* **`discmoji/context.py`**: Add `update` method in `Invoked` class to update outdated objects.
* **`discmoji/emoji.py`**: Add `update` method in `Emoji` class to update outdated objects.
* **`discmoji/gateway.py`**: Add `update` method in `GatewayManager` class to update outdated objects.
* **`discmoji/guild.py`**: Add `update` method in `Guild` class to update outdated objects.
* **`discmoji/member.py`**: Add `update` method in `GuildMember` class to update outdated objects.
* **`discmoji/message.py`**: Add `update` method in `Message` class to update outdated objects.
* **`discmoji/overwrites.py`**: Add `update` method in `ChannelPermissionOverwrite` class to update outdated objects.
* **`discmoji/roles.py`**: Add `update` method in `Role` class to update outdated objects.
* **`discmoji/team.py`**: Add `update` method in `Team` class to update outdated objects.
* **`discmoji/user.py`**: Add `update` method in `User` class to update outdated objects.

### CODE REVIEW

**Comment by mojidev-py** [comment]
??? I didn't mean like `update()` as a function, I meant update the fields?

**Comment by mojidev-py** [comment]
I feel like this might be AI

**Comment by outmaneuver** [comment]
> I feel like this might be AI

yes yes very ai .. its not like you put clear goals to your todos

**Comment by almostDemoPy** [comment]
\:skull:

File: discmoji/_http.py [hunk 1]
```diff
@@ -75,4 +75,25 @@ async def send_request(self,method: Literal['get','post','put','patch','delete']
                         warnings.warn(DiscmojiRatelimit(f"{check[1]}"))
                     return Payload(OPCODES.HTTP,d=json.loads(decoded),s="HTTP_REQUEST_RECIEVED")
 
-    
+    async def update_object(self, method: Literal['get', 'post', 'put', 'patch', 'delete'], route: str, data: dict) -> Payload:
+        async with self.httpclient as client:
+            match method:
```
Comment (almostDemoPy) [inline discmoji/_http.py:80]: you could just use a mapping here

File: discmoji/_http.py [hunk 2]
```diff
@@ -75,4 +75,25 @@ async def send_request(self,method: Literal['get','post','put','patch','delete']
...
+                    sent = await client.patch(self.base_url + route, json=data)
+                case "delete":
+                    sent = await client.delete(self.base_url + route)
+                case _:
+                    raise ValueError("Invalid HTTP method")
+            
+            parsed = await sent.read()
+            decoded = await parsed.decode(encoding="utf-8")
+            check = await self.ratelimited(sent)
+            if check[0] == True:
```
Comment (almostDemoPy) [inline discmoji/_http.py:97]: ` if check[0]: ` \:shrug:

File: discmoji/bot.py [hunk 3]
```diff
@@ -90,9 +90,16 @@ def _all_guilds_setter(self):
             self.all_guilds: List[Guild] = []
             self.all_guilds.append(Guild(self._gateway_client.current_payload.data))
     
-    
-    
+    async def update_all_objects(self):
```
Comment (almostDemoPy) [inline discmoji/bot.py:93]: this is awful. just update real-time. outline how your caching will work \:shrug:

File: discmoji/channel.py
Comment (almostDemoPy) [inline discmoji/channel.py:1]: update real-time

File: discmoji/command.py [hunk 4]
```diff
@@ -27,4 +27,9 @@ def __call__(self, function : Callable | None = None) -> None:
         return self # allows the function ( command object ) to still be used later on in the code
 
     def error(self, function: Callable) -> None:
-        self.__error_handlers += function
\ No newline at end of file
+        self.__error_handlers += function
+
+    async def update(self):
```
Comment (almostDemoPy) [inline discmoji/command.py:32]: you ca rename this to ` get_commands ` or something

**Review by almostDemoPy — COMMENTED** [review]
update real-time, that is within the object methods. you shouldn't be updating manually and iterably as that is spammy to the API. cache it instead

**Comment by XenonPy** [comment]
Sorry @almostDemoPy... I don't want a turf war between @mojidev-py and @outmaneuver on python principles lol

**Pull request closed without merging**